src/DepositHandler.sol

```diff
@@ -23,15 +23,20 @@ import {IDepositHandlerErrors} from "./interfaces/ICustomErrors.sol";
 contract DepositHandler is Pausable, AccessControl, IDepositHandlerErrors {
     bytes32 public constant MANAGER = keccak256("MANAGER");
     uint256 public immutable depositAmount;
-    uint256 public immutable bootcampDuration; // store value in seconds.
+    uint256 public immutable bootcampDuration;
     uint256 public immutable bootcampStartTime;
-    IERC20 public immutable depositToken;
-    //mapping(address => bool) public bootcampCompleted;
+    IERC20 public immutable depositToken;    
     mapping(address => depositInfo) public deposits;
 
+    enum Status { // status of the bootcamp participant. 
+        Deposit, // participant allowed for deposit
+        InProgress, // participant passing a bootcamp
+        Withdraw, // praticipant allowed for withdraw
+        Pass // bootcamp passed
```

Comment:
what is going to be assigned to a user that has deposited but not finished succesfully?

Reply:
Good question. In my opinion this person will stay in "InProgress" status at the moment and won't proceed to "Withdraw" status. Then Encode will contact him with the details and explain why he is not allowed to withdraw.
I understand, this is not the best workflow, but feel free to propose smth.

A quick solution I have at the moment is:
1. Add new status `NotPassed` or `Fail`. 
2. Create a simple function which will receive an array of addresses which are not passed a bootcamp successfully, and inside a function assign them this status `NotPassed`. In this way users will see on the ui that they are not passed a bootcamp and at least they'll understand that it's time to reach a manager and ask a question about the reasons of failure.
This would allow the contract to distinguish between participants who are allowed to withdraw their deposits (those who passed) and those who are not (those who failed).

Reply:
yeah going with Fail/NotPassed sounds good. we should distinguish between that an InProgress as using InProgress for this scenario can be misleading. We should still allow admin override for the new status so if they contact Encode and clear up their circumstances, they will be able to get their deposit back.

Reply:
`We should still allow admin override for the new status so if they contact Encode and clear up their circumstances, they will be able to get their deposit back` - I think we can solve this with extra withdraw function we discussed previously. Once user will proof Encode about circumstances, manager will allow him to just withdraw his funds, and that's it. Need to think how correctly implement this and distinguish emergency withdraw with such cases